### DISCUSSION TITLE
supabase.auth.user get { user: null } even after logged in

### DISCUSSION THREAD

**Discussion by monkeikawashima** [supabase/supabase#20398] [Questions]
### **What I want to achieve**
_When I submit a article, I need to INSERT my [**userid**] from [**profiles**] table to the
[**userid**] of [**posts**] table._

_Please help me!!_

### **Code:**
```
"use client";
import { useEffect } from "react";
import supabase from "@/utils/supabase/client";

export default function Page() {
  useEffect(() => {
    const insertPost = async () => {
      const user = supabase.auth.user();
      const { data: profileData, error: profileError } = await supabase
        .from("profiles")
        .select("userid")
        .eq("id", user?.id)
        .single();

      if (profileError) {
        console.error(profileError);
        return;
      }

      const { data: postData, error: postError } = await supabase
        .from("posts")
        .insert([{ userid: profileData?.userid }]);

      if (postError) {
        console.error(postError);
        return;
      }

      console.log(postData);
    };
    insertPost();
  }, []);

  return <div>...</div>;
}
```

<img width="836" alt="スクリーンショット 2024-01-13 14 44 45" src="https://github.com/supabase/supabase/assets/4121802/f12a6c7c-f46b-4438-96e8-2be0fc5f885d">

**Comment by Spherrrical** [comment]
First, if you look at the documentation, there is no call for `supabase.auth.user`, its **`supabase.auth.getUser()`**
Second, you need to use `data` to declare your instance like this and use await:
```jsx
const { data } = await supabase.auth.getUser();
```
I also recommend calling this at the top of your function on the inside if you'll be using it much, that way you don't have to keep calling it over and over for each function, but your entire function will have to be asynchronous then.

And then you can use `data.user?.id` instead of `user?.id` like this:
```jsx
.eq("id", data.user?.id)
```
Hope this helps!

**Reply by monkeikawashima** [reply]
@Spherrrical 
Thank you for your advise!!
Finally, I could fix the code.

Thank you so much!!
